scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -1432,28 +1433,52 @@ class Router(formatOps: FormatOps) {
           style.continuationIndent.extendSite,
           enumCase.inits.length > 1
         )
+
       // Template
-      case FormatToken(_, right @ T.KwExtends(), _) =>
+      case FormatToken(_, right, _)
+          if (right.is[T.KwExtends] || right.is[soft.KwDerives]) =>
         val template = defnTemplate(rightOwner)
-        val lastToken = template
-          .flatMap(templateCurly)
-          .orElse(template.map(_.tokens.last))
-          .getOrElse(rightOwner.tokens.last)
+        val derivesOrCurly = template
+          .flatMap { t =>
+            if (right.is[soft.KwDerives])
+              templateCurly(t)
+            else
+              templateDerivesOrCurly(t)
+          }
+
+        // we should expire before derives or else we will get additional indentation
+        val (lastToken, when) = derivesOrCurly match {
+          case Some(tk) => (tk, ExpiresOn.Before)
+          case None =>
+            val last = template
+              .map(_.tokens.last)
+              .getOrElse(rightOwner.tokens.last)
+
+            (last, ExpiresOn.After)
+        }
         binPackParentConstructorSplits(
           template.toSet,
           lastToken,
           style.continuationIndent.extendSite,
-          template.exists(_.inits.length > 1)
+          template.exists(_.inits.length > 1),
+          when
         )
+
       // trait A extends B, C, D, E
       case FormatToken(T.Comma(), right, _) if leftOwner.is[Template] =>
-        val template = leftOwner.asInstanceOf[Template]
-        val prevOwner = prevNonComment(prev(formatToken)).meta.leftOwner
+        val template: Template = leftOwner.asInstanceOf[Template]
+        val previousToken = prevNonComment(prev(formatToken))
+        val prevOwner = previousToken.meta.leftOwner
```

Comment:
looks like we didn't need to change these lines. `previousToken` is not used, and others are equivalent.

Reply:
Good catch, I had a lot of different approaches, so that probably stems from that.